e2e/single-cluster/gitrepo_test.go

```diff
@@ -369,6 +370,184 @@ var _ = Describe("Monitoring Git repos via HTTP for change", Label("infra-setup"
 	})
 })
 
+var _ = Describe("Monitoring Git repos via HTTPS with a custom CA for change", Label("infra-setup"), func() {
+	var (
+		tmpDir           string
+		clonedir         string
+		k                kubectl.Command
+		gh               *githelper.Git
+		clone            *git.Repository
+		repoName         string
+		inClusterRepoURL string
+		gitrepoName      string
+		r                = rand.New(rand.NewSource(GinkgoRandomSeed()))
+		targetNamespace  string
+	)
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+	})
+
+	JustBeforeEach(func() {
+
+		// Build git repo URL reachable _within_ the cluster, for the GitRepo
+		host, err := githelper.BuildGitHostname(env.Namespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		addr, err := githelper.GetExternalRepoAddr(env, HTTPSPort, repoName)
+		Expect(err).ToNot(HaveOccurred())
+		addr = strings.Replace(addr, "http://", "https://", 1)
```

Comment:
Question: 
Are these tests the same than the HTTP ones except for this line? 
(I haven't checked 100% but both blocks look very similar)
If so... maybe better to just add the `http/https` as a variable and reuse the code?

Reply:
The loop stuff is cool! thanks!